tests/Sabre/Xml/ParserTest.php

```diff
@@ -0,0 +1,587 @@
+<?php
+
+namespace Sabre\Xml;
+
+/**
+ * Test the XML reader.
+ * -Test with valid document
+ * -Test XML schema validation
+ * -Test invalid documents (missing start and end tag, malformed start and end tag and malformed quotes)
+ *
+ *
+ * PHP version 5
+ *
+ * Class ParserTest
+ * @author     Daan Biesterbos <daanbiesterbos@gmail.com>
+ * @license http://sabre.io/license/ Modified BSD License
+ * @since     1.0.0
+ */
+class ParserTest extends \PHPUnit_Framework_TestCase
```

Comment:
Hi Evert, 
My idea was to write a test to parse various valid and invalid formats. I also used this for development. I could refactor this and make it more compact. I'm not sure about the name though. I removed the schema validation entirely. The goal was to get rid of the eternal loops, and this test assures they do not occur.  That solution seems to be pretty solid. (and a lot faster than schema validation! :+1: )  What do you think about "ReadInvalidDocumentTest" - as most of the tests are all about testing invalid xml anyway.  A compact version of the test on top (valid format) could be moved to another test case. (or removed entirely if you feel like testing valid formats is covered by the other tests?)